trie-db/src/triedbmut.rs

```diff
@@ -131,20 +131,21 @@ where
 	fn into_encoded<F, C, H>(self, mut child_cb: F) -> Vec<u8>
 	where
 		C: NodeCodec<H>,
-		F: FnMut(NodeHandle<H::Out>) -> ChildReference<H::Out>,
+		F: FnMut(NodeHandle<H::Out>, &NodeKey) -> ChildReference<H::Out>,
 		H: Hasher<Out = O>,
 	{
 		match self {
 			Node::Empty => C::empty_node(),
 			Node::Leaf(partial, value) => C::leaf_node(&partial, &value),
-			Node::Extension(partial, child) => C::ext_node(&partial, child_cb(child)),
+			Node::Extension(partial, child) => C::ext_node(&partial, child_cb(child, &partial)),
 			Node::Branch(mut children, value) => {
 				C::branch_node(
 					// map the `NodeHandle`s from the Branch to `ChildReferences`
 					children.iter_mut()
 						.map(Option::take)
-						.map(|maybe_child|
-							maybe_child.map(|child| child_cb(child))
+						.enumerate()
+						.map(|(i, maybe_child)|
+							maybe_child.map(|child| child_cb(child, &NibbleSlice::new(&[i as u8]).encoded(false)))
```

Comment:
From what I understand of the issue, I would scramble with full key instead of partial key (just moving cursor over the key nibble slice while going down the tree). I am afraid of the case where node are the same with same partial key to.

Reply:
That's already the case. `child_cb` accepts node's partial key and combines it with parent's full key and passes it down the line.

Reply:
:+1: I missed that, smart (the cursor thing does not work for full trie recursive calculation I forgot it is done on commit).